src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,75 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_SIZE_OF_LOGIN = 6;
+    private static final int MIN_SIZE_OF_PASSWORD = 6;
+    private static final int MIN_ALLOWED_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkNullUser(user);
+        checkNullLogin(user);
+        checkNullPassword(user);
+        checkNullAge(user);
+        checkAlreadyCreatedLogin(user);
+        checkLoginLessThan6Chars(user);
+        checkPasswordLessThan6Chars(user);
+        checkNotAllowedAge(user);
+        return storageDao.add(user);
+    }
+
+    public void checkNullUser(User user) {
```

Comment:
For the future, methods that use only in class should be private 
```suggestion
    private void checkNullUser(User user) {
```

Reply:
ok I forgot about that. 'facepalm' ))